tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/PrettyWriter.scala

```diff
@@ -32,24 +33,42 @@ class PrettyWriter(writer: PrintWriter, textWidth: Int = 80, indent: Int = 2) ex
   // the stack of lambda declarations
   private var lambdaStack: List[TlaOperDecl] = Nil
 
-  def write(mod: TlaModule): Unit = {
-    writer.write(pretty(toDoc(mod), textWidth).layout)
-  }
+  def prettyWriteDoc(doc: Doc): Unit = writer.write(pretty(doc, textWidth).layout)
 
-  def write(decl: TlaOperDecl): Unit = {
-    writer.write(pretty(toDoc(decl), textWidth).layout)
-  }
+  def blockComment(commentStr: String): Doc = text(s"(* $commentStr *)") <> line
+  def lineComment(commentStr: String): Doc = text(s"\\* $commentStr") <> line
+
+  def writeBlockComment(commentStr: String): Unit = prettyWriteDoc(blockComment(commentStr))
+  def writeLineComment(commentStr: String): Unit = prettyWriteDoc(lineComment(commentStr))
+
+  def write(mod: TlaModule): Unit = prettyWriteDoc(toDoc(mod))
+
+  // Declarations have a trailing empty line
+  def write(decl: TlaOperDecl): Unit = prettyWriteDoc(toDoc(decl) <> line <> line)
 
-  def write(expr: TlaEx): Unit = {
-    writer.write(pretty(toDoc((0, 0), expr), textWidth).layout)
+  def write(expr: TlaEx): Unit = prettyWriteDoc(toDoc((0, 0), expr) <> line)
+
+  def moduleNameDoc(name: String, nDashes: Int = 5): Doc =
```

Comment:
Why do you have to pass `nDashes` and who should come up with that? Why don't you compute `nDashes` from `textWidth` and `name.length()`?